examples_tests/10.AllocatorTest/main.cpp

```diff
@@ -0,0 +1,478 @@
+	const std::uniform_int_distribution<uint32_t> alignOffsetPerFrameRange = std::uniform_int_distribution<uint32_t>(0u, maxAlignOffset);
+	const std::uniform_int_distribution<uint32_t> blockSizePerFrameRange = std::uniform_int_distribution<uint32_t>(1, maxBlockSize);
+};
+
+template<typename AlctrType>
+class AllocatorHandler
+{
+	using Traits = core::address_allocator_traits<AlctrType>;
+public:
+	
+	AllocatorHandler(AllocationCreationParameters& allocationCreationParameters)
+		: creationParameters(allocationCreationParameters) 
+	{
+	}
+	
+	void executeAllocatorTest()
+	{
+		for (size_t i = 0; i < creationParameters.multiAllocCnt; ++i)
+			executeForFrame();
+	}
+
+private:
+	struct AllocationData
+	{
+		uint32_t outAddr = AlctrType::invalid_address;
+		uint32_t size = 0u;
+		uint32_t align = 0u;
+	};
+
+	struct RandParams
+	{
+		uint32_t maxAlign;
+		uint32_t addressSpaceSize;
+		uint32_t alignOffset;
+		uint32_t offset;
+		uint32_t blockSz;
+	};
+
+private:
+	void executeForFrame()
+	{
+		RandParams randAllocParams = getRandParams();
+		void* reservedSpace = nullptr;
+		AlctrType alctr;
+
+		if constexpr (std::is_same<AlctrType, core::LinearAddressAllocator<uint32_t>>::value)
+		{
+			alctr = AlctrType(nullptr, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize);
+		}
+		else
+		{
+			const auto reservedSize = AlctrType::reserved_size(randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+			reservedSpace = _IRR_ALIGNED_MALLOC(reservedSize, _IRR_SIMD_ALIGNMENT);
+			alctr = AlctrType(reservedSpace, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+		}
+
+		// randomly decide how many `multi_allocs` to do
+		const uint32_t multiAllocCnt = rng.getRandomNumber(1u, 5u); //TODO: will change it later
+		for (uint32_t i = 0u; i < multiAllocCnt; i++)
+		{
+			outAddresses.clear();
+			sizes.clear();
+			alignments.clear();
+
+			// randomly decide how many allocs in a `multi_alloc` NOTE: must pick number less than `traits::max_multi_alloc`
+			constexpr uint32_t upperBound = core::address_allocator_traits<AlctrType>::maxMultiOps;
+			const uint32_t allocCntInMultiAlloc = rng.getRandomNumber(1u, upperBound);
+
+			for (size_t i = 0u; i < allocCntInMultiAlloc; ++i)
+			{
+				// randomly decide sizes (but always less than `address_allocator_traits::max_size`)
+				outAddresses.emplace_back(AlctrType::invalid_address);
+
+				if constexpr (std::is_same<AlctrType, core::PoolAddressAllocator<uint32_t>>::value)
+				{
+					sizes.emplace_back(randAllocParams.blockSz);
+					alignments.emplace_back(randAllocParams.blockSz);
+				}
+				else
+				{
+					sizes.emplace_back(rng.getRandomNumber(1u, Traits::max_size(alctr)));
+					alignments.emplace_back(rng.getRndMaxAlign());
```

Comment:
for the generalpurpose allocator the size needs to be >= minimum size (block size)

For Pool allocator both size and alignment should be equal.

I dont recall any restrictions on alignment for generalpurpose allocator though.

Reply:
its probably why generalpurpose crashes during the test.